### ISSUE TITLE
return -1 

### ISSUE THREAD

**Issue by kaan123123** [NadirRoGue/RenderEngine#5]
have a mem error bro.

**Comment by NadirRoGue** [comment]
Did you compile targeting x64? If you attempt to compile for x86 platform, you have to move the binaries for x86 to the main directory of the project (I configured the project to take the libraries and dlls from their respectives folders depending on the target platform, yet it doesnt seems to work in VS).

**Comment by kaan123123** [comment]
ı compiled x64 but have error :( . doesnt work in VS 2017

**Comment by NadirRoGue** [comment]
Yes, I developed it in VS 2017 Community edition. Never had this problem before, could you compile and run it on Debug mode to try to see from where this error comes?

**Comment by kaan123123** [comment]
shader = Engine::ProgramTable::getInstance().getProgram<Engine::VolumetricCloudProgram>();
I debug step step.
error here.

**Comment by NadirRoGue** [comment]
I guess you refer to this line:

`shader = Engine::ProgramTable::getInstance().getProgram<Engine::VolumetricCloudProgram>();`

Which is located in `volumetricclouds/CloudSystem.cpp`

The only reason for that -1 is a failure in loading the shader files, compile them, and link them. If you run in Debug mode again, whenever VS halts the execution, look in the engine console, as any GLSL error will appear there.

The volumetric clouds shader might be using some functionality that your hardware does not support

**Comment by kaan123123** [comment]
ı dıdnt guess GSLS error. maybe does not support machine.
I delete only vol cloud and after I will try.

**Comment by NadirRoGue** [comment]
To disable the clouds, just go to `src/skybox/SkyBox.cpp`

And comment the following lines:

* Line 16 
`clouds = new Engine::CloudSystem::VolumetricClouds();`

* Line 55
`clouds->render(camera);`

And then try to run the program.

**Comment by kaan123123** [comment]
okey ıts work and print it:
//
ThreadPool: Using 4 thread(s)
This system supports OpenGL Version: 4.1.13474 Core Profile Forward-Compatible Context 22.19.162.4
TextureTable: Maximun anisotropic level supported: 16
Generating Perlin-Worley + 3 Worley octaves volume texture (128x128x128)...
Done!
Generating 3 Worley octave volume texture (32x32x32)...
Done!
Generating Weather texture (1024x1024)...
Done!
//

but 
No Man Planet screen is white everytime. ı waited 10 minute but anything change.((

**Comment by NadirRoGue** [comment]
The noise generation is a heavy task, and I myself had to edit the windows registry to allow the GPU to execute long programs without crashing.

So for a last attempt, lets disable such noise generation (which is used only by the clouds you already disabled).

Go to src/renderers/DeferredRenderer.cpp

In line 135, comment the following line:
`Engine::CloudSystem::NoiseInitializer::getInstance().render();`

**Comment by kaan123123** [comment]
All of thank you. Program is succesfully work but my machine very bad  slow etc. ;) 
i wish you continued success in your career.
peace be with you
--
